### ISSUE TITLE
Writing non-contiguous arrays is broken

### ISSUE THREAD

**Issue by stuarteberg** [constantinpape/z5#24]
Writing a non-contiguous array (e.g. myvol[::2]) writes the wrong data to the file.  I don't know if this is true for all non-contiguous arrays, but here's my test case:

```python
grayscale = np.zeros((256, 256, 256), np.uint8)

f = z5py.File(z5_vol_path, use_zarr_format=False)

# Contiguous
ds = f.create_dataset('grayscale', dtype='uint8', shape=(256, 256, 256), chunks=(64,64,64), compressor='raw')
ds[:] = grayscale
assert (ds[:] == grayscale).all() # so far, so good.

# Non-contiguous
ds2 = f.create_dataset('downsampled', dtype='uint8', shape=(128, 128, 128), chunks=(64,64,64), compressor='raw')
ds2[:] = grayscale[::2, ::2, ::2]
assert (ds2[:] == grayscale[::2, ::2, ::2]).all(), "uh-oh"
```

If I use a contiguous array, the assertion passes:

```python
ds2[:] = grayscale[::2, ::2, ::2].copy('C')
assert (ds2[:] == grayscale[::2, ::2, ::2]).all()
```

**Comment by constantinpape** [comment]
Thanks for pointing that out Stuart.
I am pretty sure this is a side effect of the copy function I have implemented 
when reading / writing multiarrays to buffers internally.
Until now, I didn't think about this, but it is fairly obvious that what I am doing will not work for non-contigouous arrays.

I will have a look if using the (slower) built-in xtensor functions can deal with this and if they
do provide this as a fall-back for non-contiguous arrays.

**Comment by constantinpape** [comment]
FIxed in 2f5a395eb29ce06ade463f35c7893af3c74452ae .
I solved this via `np.require(data, requirements='C')` in the pythonbindings.
